### ISSUE TITLE
Deployment: Add Terraform deployment to ACA

### ISSUE THREAD

**Issue by stuartleeks** [microsoft/aoai-api-simulator#17]
Alongside the existing option to deploy to ACA using bicep, add the ability to deploy to ACA using Terraform

- add bicep and scripts 
- update docs

**Comment by martinpeck** [comment]
Adding additional notes here to validate the requirements.

Currently there are two bicep scripts:

`base.bicep` deploys the following into a resource group...

- container registry
- keyvault
- storage account

`main.bicep` then refers to these resources, and deploys...

- log analysitcs
- app insights
- managed identity
- container app env
- container app storage
- several keys to KV
- the container app

The sequence of steps for a deployment are:

1. deployment is kicked off by running `make deploy-aca`
2. this calls `./scripts/deploy-aca.sh`
3. `deploy-aca.sh` in turn, invokes the following scripts:
    - deploy-aca-base.sh
    - docker-build-and-push.sh
    - deploy-aca-infra.sh
4. Finally, `deploy-aca.sh` tests the deployment by calling `deploy-aca-test.sh` and logs from the ACA are shown by calling `deploy-aca-show-logs.sh`

So, given this, I believe the sequence of tasks related to this issue are:

- Create a "switch" to allow a developer to choose between Bicep and Terraform. Suggest env var `SIMULATOR_DEPLOY_METHOD` with values `bicep` and `terraform` as values, and `bicep` as the default
- Rename `deploy-aca-base.sh` to `deploy-aca-base-bicep.sh`
- Rename `deploy-aca-infra.sh` to `deploy-aca-infra-bicep.sh`
- Create subfolder of `./infra` as  `./infra/bicep` and move the above two files into this location
- Create subfolder of `./infra` as  `./infra/terraform`
- two new files called `deploy-aca-base-terraform.sh` and `deploy-aca-infra-bicep.sh` would be added to the `./scripts` folder, next to their bash equivalents.
- Modify `deploy-aca.sh` to inspect `SIMULATOR_DEPLOY_METHOD` and then call the correct bash scripts.
- Test that the bicep deployment continues to work when  `SIMULATOR_DEPLOY_METHOD=bicep`, or when the env var is not set, and that the terraform deployment fails when `SIMULATOR_DEPLOY_METHOD=terraform` (due to the bash scripts being empty at this stage)
- the terraform scripts need to be authored and tested manually. A set of .tf files would be added to `./infra/terraform`
- the bash scripts `deploy-aca-base-terraform.sh` and `deploy-aca-infra-bicep.sh` would need to be authored

Does this sequence of tasks look right to you @stuartleeks?

Additional Questions/Notes:

* The deployment scripts pass "values" to each other via the output JSON file. I don't see why this should be a problem, but is there any difference between the JSON output from the bicep deployment vs a terraform deployment?
* Will we then test both of these deployment mechanisms in the CI builds?

**Comment by martinpeck** [comment]
Do we think it might be better to put the terraform and bicep specific bash files into the same folder structure as the terraform and bicep scripts?

e.g. something like...

```
.
└── infra
    ├── bicep
    │   ├── base.bicep
    │   ├── main.bicep
    │   └── scripts
    │       ├── deploy-aca-base-bicep.sh
    │       └── deploy-aca-infra-bicep.sh
    └── terraform
        ├── base.tf
        ├── main.tf
        └── scripts
            ├── deploy-aca-base-terraform.sh
            └── deploy-aca-infra-terraform.sh
```

**Comment by stuartleeks** [comment]
For the Terraform deployment, we may not need the base/infra split that bicep has because we can use localexec etc in the Terraform

**Comment by martinpeck** [comment]
Yep. I'd not given much thought to the .tf files needed, so consider those as just illustations.

**Comment by mluker** [comment]
@stuartleeks  please assign me

**Comment by stuartleeks** [comment]
Closed by #60
